pkg/api/types.go

```diff
@@ -341,14 +342,75 @@ type Result struct {
 	BuildInfo BuildInfo
 }
 
-// BuildInfo holds information about a particular step in the build process.
+// BuildInfo contains information about the build process.
 type BuildInfo struct {
+	// Steps contains details about each build step.
+	Steps Steps
+
 	// FailureReason is a camel case reason that is used by the machine to reply
 	// back to the OpenShift builder with information why any of the steps in the
-	// build, failed.
+	// build failed.
 	FailureReason FailureReason
 }
 
+// Steps is a map of build steps that have been recorded
+type Steps map[StepName]StepInfo
```

Comment:
ordering makes sense... i'm less convinced on storing the duration because in general i prefer to store the raw values so we can present the data in any form we want.  Among other things, it might prove useful if we have "gaps" in the duration (since we aren't measuring absolutely everything, we could have a build that takes 60s, but steps that only account for 40s.  Being able to see when each step started/stopped could help us find the "missing" 20s).

Obviously we'd present duration values to the user, but the api object would store the actual date/time information.